### DISCUSSION TITLE
Implementing a "Red flag" for accessing data sources

### DISCUSSION THREAD

**Discussion by elsalant** [fybrik/fybrik#565] [Ideas]
In all three use cases in FogProtect, there is a scenario where the security environment of a running application is compromised.  When this is detected, there needs to be a way of blocking or restricting an M4D application's access to data.
There are (currently envisioned) three different states that the application can be in:
1. SAFE 
2. RESTRICTED 
3. LOCKDOWN
where the rego policy will dictate how to handle each situation.

What is the best way of handling this?  Maybe adding an "appState" field to the M4DApplication which will get passed to the Policy Mgr?

**Comment by simanadler** [comment]
@elsalant Let me make sure I understand.  You are saying that the workload for which the M4D environment was created is evaluated occasionally (by a component external to M4D) after the M4D environment is created, and is given the state SAFE, RESTRICTED or LOCKDOWN?  Who/what is generating the M4DApplication yaml in your scenario?  My initial thought is that it doesn't make sense to have the workload security state be part of M4DApplication yaml since this is something the user would have access to and could change.

Perhaps the workload security state would be provided by whoever/whatever is evaluating the workload, and this would be an additional source of input for the smart blueprint. https://github.com/IBM/the-mesh-for-data/issues/93.  @roee88 @ronenkat your thoughts?

**Comment by elsalant** [comment]
Yes, the workload is initially deployed in a SAFE state, but this can change during its lifecycle due to some external trigger.
The M4DApplication in the FogProtect use cases will be deployed by someone to set up an environment for a user.  For example, in one use case, the M4DApplication will set up REST module which comes between the organization's front-end dashboard and backend data store.  In this case, it makes sense to me that the same 'someone' who initially deploys the M4DApplication can change this new value, although this can also be done by some system-level monitoring application.

**Comment by roee88** [comment]
If access to the M4DApplication is restricted and trusted then you can already add this to the appInfo map and code your policies to take this into account. If it's not restricted but can be then make the change (e.g. using RBAC, gatekeeper, etc.).

Otherwise, adding this would require federated workload identities and passing the workload identity to the policy manager. This requires a design (is the identity just the selector, would that replace the location attribute, which other policy managers support this mode, etc.).

**Reply by elsalant** [reply]
@roee88 I'm willing to go for "restricted and trusted".  Thanks...
